sky/skylet/providers/kubernetes/node_provider.py

```diff
@@ -209,10 +210,42 @@ def create_node(self, node_config, tags, count):
         start = time.time()
         while True:
             if time.time() - start > self.timeout:
-                raise config.KubernetesError(
+                pod_status = new_nodes[0].status.phase
+                pod_name = new_nodes[0]._metadata._name
+                events = kubernetes.core_api().list_namespaced_event(
+                    self.namespace,
+                    field_selector=f"involvedObject.name={pod_name}")
+                for event in events.items:
+                    if event.reason == 'FailedScheduling':
+                        event_message = event.message
+                        break
+                timeout_err_msg = (
                     'Timed out while waiting for nodes to start. '
                     'Cluster may be out of resources or '
                     'may be too slow to autoscale.')
+                lack_resource_msg = (
+                    'More than available {0} are requested. '
+                    'Run \'sky status\' to see the amount of {0} '
+                    'that are already in use.')
+                if event_message is not None:
+                    if pod_status == 'Pending':
+                        if 'Insufficient cpu' in event_message:
+                            raise config.KubernetesError(
+                                lack_resource_msg.format('CPUs'))
+                        if 'Insufficient memory' in event_message:
+                            raise config.KubernetesError(
+                                lack_resource_msg.format('Memories'))
+                        if 'didn\'t match Pod\'s node affinity/selector' in event_message:
+                            node_selector = pod.spec.node_selector
+                            if node_selector is not None:
```

Comment:
Is is possible to add a check if the node_selector label was in `[lf.get_label_key() for lf in kubernetes_utils.LABEL_FORMATTER_REGISTRY]`? In the future we could have other label selectors, and we'd want to show the GPU message only if GPU label failed to match.

Reply:
Updated the condition to be if `'Insufficient nvidia.com/gpu' in event_message:` instead and only raise the message when one of key value from pod's `nodeSelector` is one of `[lf.get_label_key() for lf in kubernetes_utils.LABEL_FORMATTER_REGISTRY]`

Reply:
`'Insufficient nvidia.com/gpu'` occurs in the error message when the attempted node has the requested GPU but all are exhausted. And `'didn't match Pod's node affinity/selector'` occurs in the error message when the attempted node does not have the requested GPU at all. Both cases are now handled only when `nodeSelector` key label is in `[lf.get_label_key() for lf in kubernetes_utils.LABEL_FORMATTER_REGISTRY]`